images/base/files/usr/local/bin/entrypoint

```diff
@@ -104,10 +104,17 @@ fix_cgroup() {
   fi
   echo 'INFO: detected cgroup v1'
   echo 'INFO: fix cgroup mounts for all subsystems'
-  # see: https://d2iq.com/blog/running-kind-inside-a-kubernetes-cluster-for-continuous-integration
-  # capture initial state before modifying
+  # See: https://d2iq.com/blog/running-kind-inside-a-kubernetes-cluster-for-continuous-integration
+  # Capture initial state before modifying
+  #
+  # Basically we're looking for the cgroup-path for the cpu controller for the
+  # current process. this tells us what cgroup-path the container is in.
+  # Then we collect the subsystems that are active on this path.
+  # We assume the cpu controller is in use on all node containers.
+  #
+  # See: https://man7.org/linux/man-pages/man7/cgroups.7.html
   local current_cgroup
-  current_cgroup=$(grep systemd /proc/self/cgroup | cut -d: -f3)
+  current_cgroup=$(grep -E '^[^:]*:([^:]*,)?cpu(,[^,:]*)?:.*' /proc/self/cgroup | cut -d: -f3)
```

Comment:
```
/proc/[pid]/cgroup (since Linux 2.6.24)
              This file describes control groups to which the process
              with the corresponding PID belongs.  The displayed
              information differs for cgroups version 1 and version 2
              hierarchies.

              For each cgroup hierarchy of which the process is a
              member, there is one entry containing three colon-
              separated fields:

                  hierarchy-ID:controller-list:cgroup-path

              For example:

                  5:cpuacct,cpu,cpuset:/daemons

              The colon-separated fields are, from left to right:

              1. For cgroups version 1 hierarchies, this field contains
                 a unique hierarchy ID number that can be matched to a
                 hierarchy ID in /proc/cgroups.  For the cgroups version
                 2 hierarchy, this field contains the value 0.

              2. For cgroups version 1 hierarchies, this field contains
                 a comma-separated list of the controllers bound to the
                 hierarchy.  For the cgroups version 2 hierarchy, this
                 field is empty.

              3. This field contains the pathname of the control group
                 in the hierarchy to which the process belongs.  This
                 pathname is relative to the mount point of the
                 hierarchy.
```

Reply:
(we are matching the middle field, specifically one of the comma delimited controller entries)